lib/glossarist/lutaml_model/managed_concept.rb

```diff
@@ -0,0 +1,287 @@
+      include Glossarist::Utilities::CommonFunctions
+
+      attribute :related, RelatedConcept
+      attribute :dates, ConceptDate, collection: true
+      attribute :localized_concept, :hash
+      attribute :groups, :string, collection: true
+      attribute :sources, ConceptSource
+      attribute :date_accepted, ConceptDate
+      attribute :localizations, :hash, default: -> { {} }
+      attribute :localized_concepts, :hash, default: -> { {} }
+      attribute :data, :hash
+      attribute :status, :string, values: Glossarist::GlossaryDefinition::CONCEPT_STATUSES
+      attribute :identifier, :string
+      attribute :id, :string
+      attribute :uuid, :string
+
+      yaml do
+        map :data, with: { to: :data_to_yaml, from: :data_from_yaml }
+        map :identifier, with: { to: :identifier_to_yaml, from: :identifier_from_yaml }
+        map :id, with: { to: :identifier_to_yaml, from: :identifier_from_yaml }
+        map :related, to: :related
+        map :dates, to: :dates
+        map :localized_concept, to: :localized_concept
+        map :groups, to: :groups
+        map :sources, with: { to: :sources_to_yaml, from: :sources_from_yaml }
+        map :localizations, to: :localizations
+        map :date_accepted, with: { from: :date_accepted_from_yaml, to: :date_accepted_to_yaml }
+        map :localized_concepts, to: :localized_concepts
+        map :status, to: :status
+
+        map :uuid, to: :uuid, with: { from: :uuid_from_yaml, to: :uuid_to_yaml }
+      end
+
+      def initialize(attributes = {})
+        attributes = symbolize_keys(attributes)
+        @uuid = attributes[:uuid]
+
+        data = attributes.delete(:data) || {}
+        data["groups"] = attributes[:groups]
+        data["status"] = attributes[:status]
+
+        data = symbolize_keys(data.compact)
+        data[:identifier] = data[:id] if data[:id]
+        super(slice_keys(data, managed_concept_attributes))
+      end
+
+      def id
+        @id || @identifier
+      end
+
+      def identifier
+        @id || @identifier
+      end
+
+      def id=(val)
+        @identifier = val
+      end
+
+      def identifier=(val)
+        @identifier = val
+      end
+
+      def date_accepted_from_yaml(model, value)
+        model.dates ||= []
+        model.dates << ConceptDate.of_yaml({ "date" => value, "type" => "accepted" })
+      end
+
+      def date_accepted_to_yaml(model, doc)
+        doc["date_accepted"] = model.date_accepted.date if model.date_accepted
+      end
+
+      def data_to_yaml(model, doc)
+        doc["data"] = model.data_hash
+      end
+
+      def data_from_yaml(model, value)
+      end
+
+      def uuid_to_yaml(model, doc)
+        doc["id"] = model.uuid
+      end
+
+      def uuid_from_yaml(model, value)
+        model.uuid = value if value
+      end
+
+      def uuid
+        @uuid ||= Glossarist::Utilities::UUID.uuid_v5(Glossarist::Utilities::UUID::OID_NAMESPACE, to_yaml(except: [:uuid]))
+      end
+
+      def data_hash
```

Comment:
This is very weird. We should just use the default functionality from lutaml-model?